src/pip/_vendor/distlib/_backport/misc.py

```diff
@@ -35,7 +35,8 @@ def fsencode(filename):
         if isinstance(filename, bytes):
             return filename
         elif isinstance(filename, str):
-            return filename.encode(sys.getfilesystemencoding())
+            fs_enc = sys.getfilesystemencoding() or 'utf-8'
+            return filename.encode(fs_enc)
         else:
             raise TypeError("expect bytes or str, not %s" %
                             type(filename).__name__)
```

Comment:
Same here - we would normally expect this change to be reported to the distlib project, and vendor a fixed version when it's released.

Reply:
Will revert this change.